lua/neotest-kotlin/treesitter/init.lua

```diff
@@ -54,17 +48,28 @@ M.get_all_matches_as_string = function(path, query)
 	return results
 end
 
---- This will take in a path to a file, run a treesitter query on it, and return the first match as a string.
----@param path string path to file
----@param query string treesitter query
----@return string? first_match
-M.get_first_match_string = function(path, query)
-	local results = M.get_all_matches_as_string(path, query)
-	if #results > 0 then
-		return results[1]
-	end
+---List all classes in a provided file using treesitter
+---@param file string
+---@return string[] classes
+function M.list_all_classes(file)
+	return get_all_matches_as_string(file, class_query)
+end
+
+---Get the first java package in a provided file using treesitter
+---@param file string
+---@return string? package
+function M.java_package(file)
+	return get_all_matches_as_string(file, package_query)[1]
+end
 
-	return nil
+---Uses neotest.treeistter.parse_positions to discover all namespaces/tests in a file.
+---@param file string
+---@return neotest.Tree?
+function M.parse_positions(file)
+	return neotest.treesitter.parse_positions(file, kotest_query, {
+		nested_namespaces = true,
+		nested_tests = false,
+	})
```

Comment:
Instead of exposing the raw Treesitter queries using the `init.lua` here to only expose functions that use the treesitter queries. Moved the kotest query and parse_positions in here too. Trying to encapsulate all treesitter logic in one spot